src/lib/FreeC/Backend/Coq/Converter/TypeDecl.hs

```diff
@@ -218,6 +224,71 @@ convertDataDecl (IR.DataDecl _ (IR.DeclIdent _ name) typeVarDecls conDecls) =
         rhs
       )
 
+  -- | Generates an induction scheme for the data type.
+  generateInductionScheme :: Converter Coq.Sentence
+  generateInductionScheme = localEnv $ do
+    Just tIdent <- inEnv $ lookupIdent IR.TypeScope name
+    -- Create variables and binders.
+    let generateArg argType = do
+          ident <- freshCoqQualid freshArgPrefix
+          return $ (ident, Coq.typedBinder Coq.Explicit [ident] argType)
+    (tvarIdents, tvarBinders) <- convertTypeVarDecls' Coq.Explicit typeVarDecls
+    (propIdent , propBinder ) <- generateArg
+      (Coq.Arrow (genericApply tIdent [] [] (map Coq.Qualid tvarIdents))
+                 (Coq.Sort Coq.Prop)
+      )
+    (_hIdents, hBinders) <- mapAndUnzipM (generateInductionCase propIdent)
+                                         conDecls
+    (valIdent, valBinder) <- generateArg
+      (genericApply tIdent [] [] (map Coq.Qualid tvarIdents))
+    -- Stick everything together.
+    let schemeName = Coq.bare $ fromJust (Coq.unpackQualid tIdent) ++ "_Ind"
+        binders =
+          genericArgDecls Coq.Explicit
+            ++ tvarBinders
+            ++ [propBinder]
+            ++ hBinders
+        term = Coq.Forall
+          (NonEmpty.fromList [valBinder])
+          (Coq.app (Coq.Qualid propIdent) [Coq.Qualid valIdent])
+        scheme = Coq.Assertion Coq.Definition schemeName binders term
+        proof  = Coq.ProofDefined (Text.pack "  fix H 1; intro; prove_ind.")
```

Comment:
If tactics like `prove_ind` can be shadowed by other functions, the identifier must be reserved as well. Any way there should be a constant for the name of the tactic in `FreeC.Backend.Coq.Base`.

Reply:
It seems like tactics have a different namespace than functions, lemmas, etc. Therefore it seems not to be necessary to reserve the identifier.